### DISCUSSION TITLE
Email uniqueness

### DISCUSSION THREAD

**Discussion by vnen** [zitadel/zitadel#6453] [Q&A] [answered]
Did I miss something or is it not possible to have unique email addresses (that is, only one user per email)? The docs do say that there can be multiple users with the same email address, but I do wonder if there's a way to forbid this.

For my case I would like to use email and username interchangeably (for login) so enforcing the email uniqueness would be important.

**Comment by hifabienne** [comment]
Hei @vnen 
The user name field, has to be unique. So if you are saving the email address in the username field it has to be unique.
But the username and the email could differ as the email is used to notify users (e.g password reset), and therefore it doesn't matter if it is unique.

**Reply by vnen** [reply]
I'm not saving the email in the username field because I want custom usernames as well, so I only save the email on the email field. It does matter to me because I don't want users to register duplicate accounts either intentionally or accidentally.

I've often have been reminded I already have an account in some system because of the email address. So if a user is trying to register again using the same email (but a different username) I want this to be blocked too so they can just recover the old account instead of making a new one.

**Reply by hifabienne** [reply, marked as answer]
Thanks for the information. We are currently discussing user schemas internally. I just added the question about defining uniqueness of fields in there. https://github.com/zitadel/zitadel/issues/6433

**Reply by vnen** [reply]
Great! I appreciate the quick reply.

**Comment by vnen** [comment]
For now, since I'm adding my custom UI I'm just searching if the email is already in use before registering the user. It's not ideal (needs an extra request) but it works.
